From 2ad6d3e9325a39333be0c4b9094e9d5e756dea65 Mon Sep 17 00:00:00 2001
From: Kacper <kornet@tuptus>
Date: Thu, 9 Dec 2010 18:57:06 +0000
Subject: [PATCH] Preserve Entry class key bindings for SHA id

If bind is specified for key without any modifier, then any combination
of modifiers may be present in the event. So bind $e $ev "$escript;
break" breaks some useful bindings from Entry class (for example
Ctrl+k).
---
 gitk |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git gitk-git/gitk gitk-git/gitk
index 45e3380..0c6f3af 100755
--- gitk-git/gitk
+++ gitk-git/gitk
@@ -2528,6 +2528,10 @@
         set ::BM "2"
     }
 
+    foreach e $entries {
+        bindtags $e [linsert [bindtags $e] 2 entrybind]
+    }
+
     bind .ctop <Map> {
         bind %W <Map> {}
         %W sashpos 0 $::geometry(topheight)
@@ -2759,12 +2763,8 @@
 proc bindkey {ev script} {
     global entries
     bind . $ev $script
-    set escript [bind Entry $ev]
-    if {$escript == {}} {
-        set escript [bind Entry <Key>]
-    }
     foreach e $entries {
-        bind $e $ev "$escript; break"
+        bind entrybind $ev "break"
     }
 }
 
